src/ngraph/runtime/cpu/cpu_builder.hpp

```diff
@@ -204,6 +204,73 @@
+    if (R2 == 1)                                                                                   \
+    {                                                                                              \
+        SELECT_RANK1(KV, ET, R1, 1, K);                                                            \
+    }                                                                                              \
+    else if (R2 == 2)                                                                              \
+    {                                                                                              \
+        SELECT_RANK1(KV, ET, R1, 2, K);                                                            \
+    }                                                                                              \
+    else if (R2 == 3)                                                                              \
+    {                                                                                              \
+        SELECT_RANK1(KV, ET, R1, 3, K);                                                            \
+    }                                                                                              \
+    else if (R2 == 4)                                                                              \
+    {                                                                                              \
+        SELECT_RANK1(KV, ET, R1, 4, K);                                                            \
+    }                                                                                              \
+    else if (R2 == 5)                                                                              \
+    {                                                                                              \
+        SELECT_RANK1(KV, ET, R1, 5, K);                                                            \
+    }                                                                                              \
+    else if (R2 == 6)                                                                              \
+    {                                                                                              \
+        SELECT_RANK1(KV, ET, R1, 6, K);                                                            \
+    }                                                                                              \
+    else if (R2 == 7)                                                                              \
+    {                                                                                              \
+        SELECT_RANK1(KV, ET, R1, 7, K);                                                            \
+    }                                                                                              \
+    else                                                                                           \
```

Comment:
do you know why we expand these expressions instead of just 
```if (R2 > 7) throw error else pass R2 to SELECT_RANK1 ```

Reply:
R1 and R2 are not constexpr thus can't be used in constant expression. We can't do KV= K<ET, R1, R2>.

Reply:
yes that's what i thought. I get it now. By doing it in levels SELECT_RANK2 -> SELECT_RANK1 and changing the R2 to constexpr R2. R2 or R1 in these are not the original variables passed.